lib/rubanok/dsl/matching.rb

```diff
@@ -64,16 +64,31 @@ def build_method_name
         end
       end
 
+      module InstanceMethods
+        def default_match_handler(rule, params, fail_when_no_matches)
+          fail_when_no_matches = Rubanok.fail_when_no_matches if fail_when_no_matches.nil?
+          return raw unless fail_when_no_matches
+
+          fail ::Rubanok::BadValueError, <<~MSG
+            Bad value(s) for #{rule.fields.join(", ")}: #{params.slice(*rule.fields).values.join(", ")}
```

Comment:
In what case it could be plural?

Let's change the message as well:

```
Unexpected value were passed to "#{rule.fields.join(", ")" clause: #{value}
```

Also, this is a good candidate for adding a suggestion using a built-in `did_you_mean` (see [example](https://github.com/palkan/action_policy/blob/master/lib/action_policy/utils/suggest_message.rb)). We can add it later in a separate PR.